src/pages/index.js

```diff
@@ -56,53 +147,122 @@ const anyFormValidation = (settings) => {
+});
 
 const profileEditModal = new PopupWithForm({
-  popupSelector: "#profile-edit-modal",
+  popupSelector: "#profileModal",
   handleFormSubmit: (formValues) => {
-    profileUser.setUserInfo({
-      name: formValues.profileTitle,
-      description: formValues.profileDescription,
-    });
-    profileEditModal.close();
+    profileEditModal.renderLoading(true);
+    getPatch(formValues)
+      .then((res) => {
+        userInfo.setCurrentUser(res);
+        formValidators[
+          profileEditForm.getAttribute("name")
+        ].handleResetValidation(true);
+        profileEditModal.close();
+        profileEditModal.renderLoading(false);
+      })
+      .catch((err) => {
+        console.error(err);
+        profileEditModal.renderLoading(false);
+        alert(`${err}`);
+      });
   },
 });
 
 const cardEditModal = new PopupWithForm({
-  popupSelector: "#card-edit-modal",
+  popupSelector: "#cardAddModal",
   handleFormSubmit: (formValues) => {
-    renderCard({
-      name: formValues.cardTitle,
-      link: formValues.imageUrl,
-    });
-    formValidators[cardEditForm.getAttribute("name")].handleResetValidation(
-      true
-    );
-    cardEditModal.close();
+    cardEditModal.renderLoading(true);
+    pageApi
+      .addCardApi(formValues.cardTitle, formValues.imageUrl)
+      .then((res) => {
+        renderCard({ name: res.name, link: res.link });
+        // pulls 'item' from the res for renderCard, sends to the Card constructor 'data'
+        // data should include the properties: name, link, _id
+        formValidators[cardEditForm.getAttribute("name")].handleResetValidation(
+          true
+        );
+        cardEditModal.close();
+        cardEditModal.renderLoading(false);
+      })
+      .catch((err) => {
+        console.error(err);
+        cardEditModal.renderLoading(false);
+        alert(`${err}`);
+      });
   },
 });
 
-// Initializations
-anyFormValidation(validationSettings);
-cardsList.renderItems();
-formValidators[profileEditForm.getAttribute("name")].enableValidation();
-formValidators[cardEditForm.getAttribute("name")].enableValidation();
+const cardDeleteModal = new PopupWithForm({
+  popupSelector: "#cardDeleteModal",
+  handleFormSubmit: (_, cardInstance, cardId) => {
+    // params: 1. formValues (NA), 2. full card instance, 3. extracted _cardId for API
+
+    pageApi
+      .deleteCardApi(cardId)
+      .then(() => {
+        //REVIEW - do i need the res anywhere?
+        cardInstance.handleDeleteCard();
+        cardDeleteModal.close();
+      })
+      .catch((err) => {
+        console.error(err);
+        alert(`${err}`);
+      });
+  },
+});
 
 //SECTION - Listeners
 
 previewImageModal.setEventListeners();
 profileEditModal.setEventListeners();
+avatarEditModal.setEventListeners();
 cardEditModal.setEventListeners();
+cardDeleteModal.setEventListeners();
+
+avatarEditButton.addEventListener("click", () => {
+  formValidators[avatarEditForm.getAttribute("name")].handleResetValidation(
```

Comment:
You need to **disable** the button only in a `.then` block after a **successful response** to be able to click it again if there is a server error.